src/app/app-routing.module.ts

```diff
@@ -1,104 +1,123 @@
+        canActivate: [MenuGuardService],
+        canActivateChild: [MenuGuardService],
         path: 'dot-browser',
         loadChildren: 'app/portlets/dot-browser/dot-browser.module#DotBrowserModule'
     },
     {
-        canActivateChild: [RoutingPrivateAuthService],
+        canActivate: [MenuGuardService],
+        canActivateChild: [MenuGuardService],
         path: 'pl',
         loadChildren:
             'app/view/components/_common/pattern-library/pattern-library.module#PatternLibraryModule'
     },
     {
-        canActivateChild: [RoutingPrivateAuthService],
+        canActivate: [MenuGuardService],
+        canActivateChild: [MenuGuardService],
         path: 'notLicensed',
         loadChildren:
             'app/view/components/not-licensed/not-licensed.module#NotLicensedModule'
-    }
-];
-
-const PORTLETS_IFRAME: Routes = [
+    },
+    // END PORTLETS_ANGULAR
+    // START PORTLETS_IFRAME
     {
-        canActivateChild: [RoutingContentletAuthService],
-        path: '',
+        canActivateChild: [MenuGuardService],
+        path: 'c',
         children: [
             {
                 component: IframePortletLegacyComponent,
-                path: 'add/:id'
+                path: ':id'
+            },
+            {
+                path: '',
+                children: []
             }
         ]
     },
     {
-        canActivateChild: [RoutingPrivateAuthService],
-        path: '',
+        canActivateChild: [ContentletGuardService],
+        path: 'add',
         children: [
             {
                 component: IframePortletLegacyComponent,
-                path: 'c/:id'
+                path: ':id'
+            },
+            {
+                path: '',
+                children: []
             }
         ]
-    }
-];
-
-const AUTH_MODULES: Routes = [
-    {
-        path: 'forgotPassword',
-        loadChildren:
-            'app/view/components/login/forgot-password-component/forgot-password.module#ForgotPasswordModule'
-    },
-    {
-        path: 'login',
-        loadChildren: 'app/view/components/login/login-component/login.module#LoginModule'
     },
+    // END PORTLETS_IFRAME
     {
-        path: 'resetPassword/:token',
-        loadChildren:
-            'app/view/components/login/reset-password-component/reset-password.module#ResetPasswordModule'
+        canActivate: [MenuGuardService],
+        path: '',
+        children: []
     }
 ];
 
 const appRoutes: Routes = [
+
     {
-        canActivate: [RoutingPrivateAuthService],
-        children: [...PORTLETS_ANGULAR, ...PORTLETS_IFRAME],
-        component: MainComponentLegacy,
-        path: ''
-    },
-    {
-        canActivate: [RoutingPublicAuthService],
+        canActivate: [PublicAuthGuardService],
         children: AUTH_MODULES,
         component: LoginPageComponent,
         path: 'public'
     },
     {
-        canActivate: [RoutingPrivateAuthService],
+        canActivate: [AuthGuardService],
         children: [
```

Comment:
Did you test this case? you can testes by going to ```Sites > Edit a site > Open the tab Rules```

Reply:
it works. :)